app/models/mingle/twitter/tweets_fetcher.rb

```diff
@@ -0,0 +1,69 @@
+module Mingle::Twitter
+  class TweetsFetcher
+    def initialize(hashtag, options)
+      @hashtag = hashtag
+      @options = options
+    end
+
+    def fetch
+      valid_tweets.each {|t| t.save! }
+    end
+
+    private
+
+    attr_reader :hashtag, :options
+
+    def valid_tweets
+      tweets_for_hashtag.reject {|t| ignore? t}
+    end
+
+    def tweets_for_hashtag
+      tweetdata_for_hashtag.collect do |data|
+        Tweet.find_or_initialize_by(tweet_id: data.id.to_s).tap do |tweet|
+          tweet.attributes = {
+            created_at: data.created_at,
+            image_url: 
+            data.media.any? ? data.media.first.media_url_https.to_s : nil,
```

Comment:
I know the style guide says that you should keep within 80 characters, but personally I'd probably leave this with the previous line. :smiley: